pkgs/development/libraries/science/math/blis/default.nix

```diff
@@ -0,0 +1,64 @@
+{ stdenv
+, fetchFromGitHub
+, perl
+, python3
+
+# Enable BLAS interface with 64-bit integer width.
+, blas64 ? false
+
+# Target architecture. x86_64 builds Intel and AMD kernels.
+, withArchitecture ? "x86_64"
+
+# Enable OpenMP-based threading.
+, withOpenMP ? true
+}:
+
+let
+  blasIntSize = if blas64 then "64" else "32";
+in stdenv.mkDerivation rec {
+  pname = "blis";
+  version = "0.7.0";
+
+  src = fetchFromGitHub {
+    owner = "flame";
+    repo = "blis";
+    rev = version;
+    sha256 = "13g9kg7x8j9icg4frdq3wpl2cmp0jnh93mw48daa7ym399w17423";
+  };
+
+  inherit blas64;
+
+  nativeBuildInputs = [
+    perl
+    python3
+  ];
+
+  doCheck = true;
+
+  enableParallelBuilding = true;
+
+  configureFlags = [
+    "--enable-cblas"
+    "--blas-int-size=${blasIntSize}"
+  ] ++ stdenv.lib.optionals withOpenMP [ "--enable-threading=openmp" ]
+    ++ [ withArchitecture ];
+
+  postPatch = ''
+    patchShebangs configure build/flatten-headers.py
+  '';
+
+  postInstall = ''
+    ln -s $out/lib/libblis.so.3 $out/lib/libblas.so.3
+    ln -s $out/lib/libblis.so.3 $out/lib/libcblas.so.3
+    ln -s $out/lib/libblas.so.3 $out/lib/libblas.so
+    ln -s $out/lib/libcblas.so.3 $out/lib/libcblas.so
+  '';
+
+  meta = with stdenv.lib; {
+    description = "BLAS-compatible linear algebra library";
```

Comment:
Can we update the alternatives section of the manual to note this as an overlay choice?

https://hydra.nixos.org/build/126593072/download/1/nixpkgs/manual.html#sec-overlays-alternatives

Reply:
Added a commit that adds AMD BLIS to that section.

We should probably also add a warning to the MKL item (in a separate PR) that it currently does not make much sense to use on non-Intel CPUs. Except for `dgemm`, MKL will currently use slow SSE kernels on e.g. AMD Zen and the old `MKL_DEBUG_CPU_TYPE` workaround was removed in 2020.1. It's still possible to circumvent the Intel check, but it's a bit more involved now. This is all a big shame, because you can easily get a 50% performance improvement by using PyTorch with MKL, because it will use batched GEMM. With other BLAS libraries, it will just use normal GEMM and there will be no batch-level parallelization except for very small matrices.